core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/predicate/ContextualPredicate.java

```diff
@@ -272,6 +181,98 @@ private static boolean isBooleanOperator(String input) {
         return index == 0;
     }
 
+    public boolean matches(Event properties) {
+        String methodName = "matches";
+        LOGGER.debug("ENTERING: {}", methodName);
+
+        LOGGER.debug("Headers: {}", properties);
+
+        ContextualEvaluationCriteria cec = null;
+        Map<String, Object> contextualMap = (Map<String, Object>) properties
+                .getProperty(PubSubConstants.HEADER_CONTEXTUAL_KEY);
+
+        if (contextualMap == null) {
+            LOGGER.debug("No contextual metadata to search against.");
+            return false;
+        }
+
+        String operation = (String) properties.getProperty(PubSubConstants.HEADER_OPERATION_KEY);
+        LOGGER.debug("operation = {}", operation);
+        String metadata = (String) contextualMap.get("METADATA");
+        LOGGER.debug("metadata = [{}]", metadata);
+
+        // If deleting a catalog entry and the entry's metadata is only the word "deleted" (i.e.,
+        // the
+        // source is deleting the catalog entry and did not send any metadata with the delete
+        // event), then
+        // cannot apply any contextual filtering - just send the event on to the subscriber
+        if (operation.equals(PubSubConstants.DELETE) && metadata
```

Comment:
Do you think it is worth a little bit of defensive code here in case the Map doesn't have these fields?